drivers/serial/uart_renesas_ra8_sci_b.c

```diff
@@ -0,0 +1,1165 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra8_uart_sci_b
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/irq.h>
+#include <soc.h>
+#include "r_sci_b_uart.h"
+#include "r_dtc.h"
+#include "r_transfer_api.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ra8_uart_sci);
```

Comment:
The uart_sci will need support in the future.
The name should be `ra8_uart_sci_b` here.